src/unix/lwt_io.mli

```diff
@@ -360,16 +360,30 @@ val write_value : output_channel -> ?flags : Marshal.extern_flags list -> 'a ->
 
 val fprint : output_channel -> string -> unit Lwt.t
 val fprintl : output_channel -> string -> unit Lwt.t
+
 val fprintf : output_channel -> ('a, unit, string, unit Lwt.t) format4 -> 'a
+(* %! does nothing here. To flush the channel, use Lwt_io.(flush channel) *)
```

Comment:
Hey, just in case: to make this an [ocamldoc comment](https://caml.inria.fr/pub/docs/manual-ocaml/ocamldoc.html#sec349), it should be:

```
(** [%!] does nothing here. To flush the channel, use [Lwt_io.flush channel]. *)
```

The `(**` makes ocamldoc pick it up (so it is included in the generated manual), and the `[...]` is the ocamldoc source code style. I also took the liberty of not opening `Lwt_io` in `channel`, because that is likely to be a user's identifier, rather than something from `Lwt_io` when someone is using `fprintf`. But for `print*f` and `eprint*f`, it should probably be with `Lwt_io` as you have it, and `Lwt_io.stdout` and `Lwt_io.stderr`, respectively, rather than a user's identifier "`channel`".

Reply:
Sorry, but I'm not sure what you mean by "not opening `Lwt_io` in `channel`".

Reply:
Oh, the `Lwt_io.(foo bar)` syntax is a "local open," it brings the identifiers of `Lwt_io` in scope for both `foo` and `bar`.

So, for stdout, we probably want something like `Lwt_io.(flush stdout)`, because that means "take both `flush` and `stdout` potentially from `Lwt_io` (or anything else that is in scope)." 

But for a user's custom channel, `Lwt_io.flush channel`, because only `flush` here would come from `Lwt_io`.

Reply:
Ah, sorry; I just saw this comment. Thanks for the explanation.

I addressed this in 5cd987d5b99db37091e64b5c65b111311569813b. Does it look okay?

In f103d781161b9975215809e3f84bf090a0a5c7e5, I linked to `Lwt_io.flush` for the `fprint*f` docs. I tried to do the same for `print*f` and `eprint*f`, but I couldn't find a way to link to the identifiers `flush` and `stdout` in docs of the form `Lwt_io.(flush stdout)`. Am I missing something?

Reply:
Thank you, it looks good now :) For the cross-reference links, I'd say just not to use them, because, as you can see, the support for including cross-references inside expressions is lacking, and you have to manually try to guess where to insert spaces around and inside `[...]`, etc. It's difficult for most people to predict how that will be displayed, and it makes the `.mli` file hard to read.

The rule I use for myself is:

- If it's a single identifier on its own, try to get it cross-referenced with `{!...}`.
- If it's a non-trivial expression like `Lwt_io.flush foo`, don't try to get anything inside it cross-referenced, and just write it verbatim.

So, my suggestion is to omit f103d78.

Reply:
cc9328e reverts f103d78.